addons/pinniped/config-controller/hack/run.sh

```diff
@@ -5,22 +5,141 @@
-kubectl logs -n pinniped deploy/pinniped-config-controller-manager -f
+# the namespace must exist before the package can be deployed
+echo -e "${YELLOW}creating namespace ${APP_NAME}...${DEFAULT}"
+ytt --data-value "tanzuAuth.namespace=${APP_NAME}" --file "${TANZU_AUTH_CONFIG_DIR}/namespace.yaml" | kubectl apply -f -
+
+echo -e "${YELLOW}rebuilding image lock file with ytt and kbld...${DEFAULT}"
+ytt --data-value "image=${CONTROLLER_IMAGE}" --file "${TANZU_AUTH_DEFAULT_VALUES_FILE}" --file "${TANZU_AUTH_CONFIG_DIR}" \
+  | kbld --file - --imgpkg-lock-output "${TANZU_AUTH_BUNDLE_DIR}/.imgpkg/images.yml"
+
+# push the tanzu-auth-controller-manager-package to the registry
+echo -e "${YELLOW}pushing ${PACKAGE_IMAGE} to registry with imgpkg...${DEFAULT}"
+imgpkg push --bundle "${PACKAGE_IMAGE}" --file "${TANZU_AUTH_BUNDLE_DIR}"
+
+# inject the package name into the package file & deploy
+echo -e "${YELLOW}creating package ${TANZU_AUTH_PACKAGE_FILE} with image ${PACKAGE_IMAGE}...${DEFAULT}"
+yq e ".spec.template.spec.fetch[0].imgpkgBundle.image = \"${PACKAGE_IMAGE}\"" "${TANZU_AUTH_PACKAGE_FILE}" \
+  | kubectl apply -f -
+kubectl apply -f "${TANZU_AUTH_DIR}/metadata.yaml"
+
+
+# Deploy the package on the cluster
+echo -e "${YELLOW}creating package install...${DEFAULT}"
+PACKAGE_SA_NAME="${CONTROLLER_MANAGER_PACKAGE_NAME}-sa"
+PACKAGE_NAMESPACE="tkg-system"
+PACKAGE_INSTALL="$(cat <<EOF
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: ${PACKAGE_SA_NAME}
+  namespace: ${PACKAGE_NAMESPACE}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: ${PACKAGE_SA_NAME}-cluster-admin
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- kind: ServiceAccount
+  name: ${PACKAGE_SA_NAME}
+  namespace: ${PACKAGE_NAMESPACE}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ${CONTROLLER_MANAGER_PACKAGE_NAME}-config
+  namespace: ${PACKAGE_NAMESPACE}
+stringData:
+  values.yaml: |
+    tanzuAuth:
+      namespace: ${CONTROLLER_NAMESPACE_NAME}
+    image: ${CONTROLLER_IMAGE}
+---
+apiVersion: packaging.carvel.dev/v1alpha1
+kind: PackageInstall
+metadata:
+  name: ${APP_NAME}
+  namespace: ${PACKAGE_NAMESPACE}
+spec:
+  packageRef:
+    refName: ${APP_NAME}.tanzu.vmware.com
+    versionSelection:
+      constraints: "1.6.0"
+      prereleases: {}
+  syncPeriod: 30s
+  serviceAccountName: ${PACKAGE_SA_NAME}
+  values:
+  - secretRef:
+      name: ${CONTROLLER_MANAGER_PACKAGE_NAME}-config
+EOF)"
+
+echo -e "${YELLOW}package install contents...${DEFAULT}"
+echo "${PACKAGE_INSTALL}"
+
+echo -e "${YELLOW}applying package install contents...${DEFAULT}"
+echo "${PACKAGE_INSTALL}" | kubectl apply -f -
+
+echo -e "${YELLOW}waiting for reconciliation...${DEFAULT}"
+kubectl wait --timeout=1m --for=condition=ReconcileSucceeded app "${APP_NAME}" -n "${PACKAGE_NAMESPACE}"
+
+# Tail the logs
+kapp inspect --app "${APP_NAME}-ctrl" -n "${PACKAGE_NAMESPACE}"
```

Comment:
Also, some interesting things here, I felt were "gotchas" as there is overlap in terminology.  An `app` to `kapp` and an `app` to `kubectl` have different meanings:

```bash
kapp list -n tkg-system
Target cluster 'https://10.220.192.239:6443/' (nodes: tkg-mgmt-vc-control-plane-h4d28, 1+)

Apps in namespace 'tkg-system'

Name                                    Namespaces      
tanzu-auth-ctrl                         (cluster),tanzu-auth
```
vs
```bash
kubectl get app -A
NAMESPACE    NAME                                DESCRIPTION           SINCE-DEPLOY   AGE
tkg-system   tanzu-auth                          Reconcile succeeded   44s          4m35s
```
In other words, you cannot `kubectl get app tanzu-auth-ctrl -n tkg-system -o yaml` with the app name that kapp reports. 
If you `kubectl get app tanzu-auth -n tkg-system -o yaml` you will not see `tanzu-auth-ctrl` indicated anywhere. You can see:

```yaml
...
friendlyDescription: Reconcile succeeded
  inspect:
    exitCode: 0
    stdout: |-
      Target cluster '[https://100.64.0.1:443](https://100.64.0.1/)' (nodes: tkg-mgmt-vc-control-plane-h4d28, 1+)
      Resources in app 'tanzu-auth-ctrl'
      Namespace   Name                                           Kind                Owner    Conds.  Rs  Ri  Age
```
which does have `tanzu-auth-ctrl` in that text string at the bottom of the `stdout` msg.

This is because an 
- `App CR` is  k8s resource which kapp-controller recognises and has customizable stages such as fetch, template and deploy.
- a kapp App which is a representation of a set of k8s resources that all work together to form some higher level application
An `App CR` can be a part of a kapp `App`.